src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,78 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FC } from 'react';
+import { FilterType } from '../../types/FilterType';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string,
+  onQueryChange: (query: string) => void,
+  filterType: FilterType,
+  onFilterChange: (filterType: FilterType) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: FC<Props> = (props) => {
```

Comment:
Also can be wrapped into `React.memo` to re-render only if we change `query` or `filterType`, cause now it always re-renders with `App`. E.g. when select `todo` we don't want to re-render filter component. You can ensure of this behavior with `console.log('Render ${componentName}')` like I did :)

Reply:
> Also can be wrapped into `React.memo` to re-render only if we change `query` or `filterType`, cause now it always re-renders with `App`. E.g. when select `todo` we don't want to re-render filter component. You can ensure of this behavior with `console.log('Render ${componentName}')` like I did :)

cool, thanks:)